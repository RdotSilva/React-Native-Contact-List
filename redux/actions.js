// Action Types
export const UPDATE_USER = 'UPDATE_USER';
export const UPDATE_CONTACT = 'UPDATE_CONTACT';
export const LOG_IN_SENT = 'LOG_IN_SENT';
export const LOG_IN_FULFILLED = 'LOG_IN_FULFILLED';
export const LOG_IN_REJECTED = 'LOG_IN_REJECTED';

//Action Creators
export const updateUser = update => ({
	type: UPDATE_USER,
	payload: update
});

export const addContact = newContact => ({
	type: UPDATE_CONTACT,
	payload: newContact
});

// Async Action Creator
const logInUser = (username, password) => dispatch => {
	dispatch({ type: 'LOG_IN_SENT' });
	login(username, password)
		.then(() => {
			dispatch({ type: 'LOG_IN_SUCCESS' });
		})
		.catch(err => {
			dispatch({ type: 'LOG_IN_REJECTED' });
		});
};
