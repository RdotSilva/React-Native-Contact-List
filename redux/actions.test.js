import * as actions from './actions';

describe('updateUser returns actions', () => {
	it('returns an action', () => {
		expect(actions.updateUser({ name: 'test name' })).toMatchSnapshot();
	});

	it('handles empty object', () => {
		expect(actions.updateUser({})).toMatchSnapshot();
	});

	it('handles emptyy name', () => {
		expect(actions.updateUser({ name: '' })).toMatchSnapshot();
	});
});

describe('loginUser returns actions', () => {
	const mockLogin = (username, password) => {
		if (username === 'u' && password === 'p') {
			return 'thisIsATestToken';
		}
		throw new Error('incorrect creds');
	};

	it('dispatches LOG_IN_SENT', async () => {
		const mockDispatch = jest.fn();
		await actions.logInUser('', '')(mockDispatch);
		// mockDispatch.mock.calls all the args that the mock fn was invoked on
		expect(mockDispatch.mock.calls[0][0]).toEqual({
			type: actions.LOG_IN_SENT
		});
	});
	it('dispatches LOG_IN_FULFILLED with correct creds', async () => {
		const mockDispatch = jest.fn();
		await actions.logInUser('u', 'p', mockLogin)(mockDispatch);

		expect(mockDispatch.mock.calls[1][0]).toEqual({
			type: actions.LOG_IN_FULFILLED,
			payload: 'thisIsATestToken'
		});
		expect(mockDispatch.mock.calls[1]).toMatchSnapshot();
	});
});
