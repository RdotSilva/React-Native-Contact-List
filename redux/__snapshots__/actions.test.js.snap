// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateUser retruns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_USER",
}
`;

exports[`updateUser retruns actions handles emptyy name 1`] = `
Object {
  "payload": Object {
    "name": "",
  },
  "type": "UPDATE_USER",
}
`;

exports[`updateUser retruns actions returns an action 1`] = `
Object {
  "payload": Object {
    "name": "test name",
  },
  "type": "UPDATE_USER",
}
`;

exports[`updateUser returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_USER",
}
`;

exports[`updateUser returns actions handles emptyy name 1`] = `
Object {
  "payload": Object {
    "name": "",
  },
  "type": "UPDATE_USER",
}
`;

exports[`updateUser returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "name": "test name",
  },
  "type": "UPDATE_USER",
}
`;
